tests/__init__.py

```diff
@@ -359,3 +361,58 @@ def assert_url_equal(url1, url2):
     assert_equal(parts1.hostname, parts2.hostname)
     assert_equal(parts1.port, parts2.port)
     assert_equal(parse_qs(parts1.query), parse_qs(parts2.query))
+
+
+class HTTPStubberException(Exception):
+    pass
+
+
+class RawResponse(BytesIO):
+    def stream(self, **kwargs):
+        contents = self.read()
+        while contents:
+            yield contents
```

Comment:
I think you're right, but from a general correctness perspective I would rather the stream method work like this just in case that assumption ever changed.

Reply:
Right now a customer doesn't really have any way to make it behave differently since the interface asks for bytes. What about letting it take an object? That would also let customers do things like emulate a slow connection.

In any case, if you want this behavior you should have a test for it.

Reply:
Seeing as there's no real way to currently test it and the fact that this isn't a public interface yet I think it's okay to leave it as is.